filters/include/pcl/filters/impl/voxel_grid_covariance.hpp

```diff
@@ -323,8 +323,7 @@ pcl::VoxelGridCovariance<PointT>::applyFilter (PointCloud &output)
         voxel_centroids_leaf_indices_.push_back (static_cast<int> (it->first));
 
       // Single pass covariance calculation
-      leaf.cov_ = (leaf.cov_ - 2 * (pt_sum * leaf.mean_.transpose ())) / leaf.nr_points + leaf.mean_ * leaf.mean_.transpose ();
-      leaf.cov_ *= (leaf.nr_points - 1.0) / leaf.nr_points;
+      leaf.cov_ = (leaf.cov_ - pt_sum * leaf.mean_.transpose()) / (leaf.nr_points - 1.0);
```

Comment:
Hm... I'm sorry ,but I think the code here should be `leaf.cov_ = (leaf.cov_ - 2 * (pt_sum * leaf.mean_.transpose ()) + leaf.nr_points * leaf.mean_ * leaf.mean_.transpose ())  / (leaf.nr_points - 1.0);` rather than `leaf.cov_ = (leaf.cov_ - pt_sum * leaf.mean_.transpose()) / (leaf.nr_points - 1.0);`

Reply:
Here is a short derivation of the formula:
```
1/(m-1) sum((y-mean)(y-mean)^T)
1/(m-1) sum(y*y^T - mean*y^T - y*mean^T + mean*mean^T)
1/(m-1) (sum(y*y^T) - sum(mean*y^T) - sum(y*mean^T) + sum(mean*mean^T))
1/(m-1) (sum(y*y^T) - mean*sum(y^T) - sum(y)*mean^T + m*mean*mean^T)
1/(m-1) (sum(y*y^T) - m*mean*mean^T - m*mean*mean^T + m*mean*mean^T) because sum(y)=m*mean
1/(m-1) (sum(y*y^T) - m*mean*mean^T)
1/(m-1) (sum(y*y^T) - sum(y)*mean^T)
```